File: examples/hello_world.rb
require File.expand_path('../../lib/katamari', __FILE__)

class HelloWorldHandler < Katamari::Handler::Upstream
  def messageReceived(ctx, e)
    env = e.message
    env.response.body = 'Hello World!'
    env.response.end
  end
end

Katamari::Server.new.add('hello_world', HelloWorldHandler.new).listen(8080)

File: src/main/java/io/katamari/handler/HelloWorld.java
package io.katamari.handler;

import io.netty.channel.ChannelHandlerContext;

import io.katamari.Env;
import io.katamari.handler.InboundMessageHandler;

public class HelloWorld extends InboundMessageHandler {
  @Override
  public void messageReceived(ChannelHandlerContext ctx, Env env) throws Exception {
    env.getResponse().end("Hello World");
  }

  @Override
  public void endMessageReceived(ChannelHandlerContext ctx) throws Exception {
    ctx.flush();
  }
}